server/model/data_stores.go

```diff
@@ -111,6 +111,7 @@ func (ds DataStore) HasID() bool {
 }
 
 func (ds DataStore) Validate() error {
+	//TODO: replace for resource validate
```

Comment:
would it make sense to address these TODOs in this PR?

Reply:
Yes! I'm working on it.